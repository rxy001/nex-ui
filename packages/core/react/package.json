{
  "name": "@ant-ui/react",
  "version": "1.0.0",
  "author": "X1ng Yu",
  "scripts": {},
  "main": "src/index.ts",
  "dependencies": {
  }
}
