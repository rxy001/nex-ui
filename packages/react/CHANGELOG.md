# @nex-ui/react

## 0.0.15

### Patch Changes

- fdd4483: Release
- Updated dependencies [fdd4483]
  - @nex-ui/styled@0.0.15
  - @nex-ui/system@0.0.15
  - @nex-ui/hooks@0.0.15
  - @nex-ui/icons@0.0.15
  - @nex-ui/utils@0.0.15

## 0.0.14

### Patch Changes

- ca25983: Release
- Updated dependencies [ca25983]
  - @nex-ui/styled@0.0.14
  - @nex-ui/system@0.0.14
  - @nex-ui/hooks@0.0.14
  - @nex-ui/icons@0.0.14
  - @nex-ui/utils@0.0.14

## 0.0.13

### Patch Changes

- 1b97788: Release
- Updated dependencies [1b97788]
  - @nex-ui/styled@0.0.13
  - @nex-ui/system@0.0.13
  - @nex-ui/hooks@0.0.13
  - @nex-ui/icons@0.0.13
  - @nex-ui/utils@0.0.13

## 0.0.12

### Patch Changes

- 664eb21: Test

## 0.0.11

### Patch Changes

- 259356e: Test
- Updated dependencies [259356e]
  - @nex-ui/styled@0.0.11
  - @nex-ui/system@0.0.11
  - @nex-ui/hooks@0.0.11
  - @nex-ui/icons@0.0.11
  - @nex-ui/utils@0.0.10

## 0.0.10

### Patch Changes

- 5a229bd: release
- Updated dependencies [5a229bd]
  - @nex-ui/styled@0.0.10
  - @nex-ui/system@0.0.10
  - @nex-ui/icons@0.0.10

## 0.0.9

### Patch Changes

- 4a5c359: release
- Updated dependencies [4a5c359]
  - @nex-ui/styled@0.0.9
  - @nex-ui/system@0.0.9
  - @nex-ui/hooks@0.0.9
  - @nex-ui/icons@0.0.9
  - @nex-ui/utils@0.0.9

## 0.0.8

### Patch Changes

- c8c1956: release
- Updated dependencies [c8c1956]
  - @nex-ui/styled@0.0.8
  - @nex-ui/system@0.0.8
  - @nex-ui/hooks@0.0.8
  - @nex-ui/icons@0.0.8
  - @nex-ui/utils@0.0.8

## 0.0.7

### Patch Changes

- 2618e1a: release
- Updated dependencies [2618e1a]
  - @nex-ui/styled@0.0.7
  - @nex-ui/system@0.0.7
  - @nex-ui/hooks@0.0.7
  - @nex-ui/icons@0.0.7
  - @nex-ui/utils@0.0.7

## 0.0.6

### Patch Changes

- 87191c1: release
- Updated dependencies [87191c1]
  - @nex-ui/styled@0.0.6
  - @nex-ui/system@0.0.6
  - @nex-ui/hooks@0.0.6
  - @nex-ui/icons@0.0.6
  - @nex-ui/utils@0.0.6

## 0.0.5

### Patch Changes

- 7631577: release
- Updated dependencies [7631577]
  - @nex-ui/styled@0.0.5
  - @nex-ui/system@0.0.5
  - @nex-ui/hooks@0.0.5
  - @nex-ui/icons@0.0.5
  - @nex-ui/utils@0.0.5

## 0.0.4

### Patch Changes

- 70cb478: v0.0.4
- Updated dependencies [70cb478]
  - @nex-ui/styled@0.0.4
  - @nex-ui/system@0.0.4
  - @nex-ui/hooks@0.0.4
  - @nex-ui/icons@0.0.4
  - @nex-ui/utils@0.0.4
