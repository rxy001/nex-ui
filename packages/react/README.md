<div style="background:white;">
  <p align="center">
    <img src='../../assets/logo.png' style="width: 150px; height: 150px;" />
  </p>
  <h1 align="center">Nex UI</h1>
  <p align="center">ðŸŽ‰ A beautiful, modern, and reliable React component library.</p>
  <p align="center"><b>Theme Customizable, Light & Dark UI, Unique DX, Written In TypeScript</b></p>
</div>

## Documentation

Visit https://nex-ui-docs.vercel.app to view the full documentation.

## Getting Started

Visit https://nex-ui-docs.vercel.app/docs/getting-started to get started with Nex UI.

## Storybook

Visit https://nex-ui-storybook.vercel.app to view the storybook for all components.

## Contributing

Contributions are always welcome!

See [CONTRIBUTING.md](/CONTRIBUTING.md) for ways to get started.

Please adhere to this project's [CODE_OF_CONDUCT](/CODE_OF_CONDUCT.md).

## License

[MIT](https://choosealicense.com/licenses/mit/)
