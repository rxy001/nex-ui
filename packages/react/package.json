{
  "name": "@nex-ui/react",
  "version": "0.0.12",
  "author": "X1ng Yu",
  "description": "ðŸŽ‰ A beautiful, modern, and reliable React component library.",
  "homepage": "https://nex-ui-docs.vercel.app/",
  "private": false,
  "keywords": [
    "react",
    "react-component",
    "nui",
    "nex-ui"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/rxy001/nex-ui.git",
    "directory": "packages/react"
  },
  "license": "MIT",
  "scripts": {
    "build": "tsx ../../scripts/build.ts --dts",
    "build:fast": "tsx ../../scripts/build.ts",
    "typecheck": "tsc --noEmit",
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "type": "module",
  "files": [
    "dist"
  ],
  "main": "src/index.ts",
  "devDependencies": {
    "@storybook/react": "8.6.12"
  },
  "dependencies": {
    "@nex-ui/hooks": "workspace:*",
    "@nex-ui/icons": "workspace:*",
    "@nex-ui/styled": "workspace:*",
    "@nex-ui/system": "workspace:*",
    "@nex-ui/utils": "workspace:*",
    "clsx": "2.1.1"
  },
  "peerDependencies": {
    "@emotion/react": ">=11",
    "motion": "^12.7.2",
    "react": ">=19",
    "react-dom": ">=19"
  },
  "sideEffects": false,
  "clean-package": "../../clean-package.config.json"
}
