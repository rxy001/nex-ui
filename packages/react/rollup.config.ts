import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { defineConfig } from 'rollup'
import typescript from '@rollup/plugin-typescript'
import del from 'rollup-plugin-delete'
import dts from 'rollup-plugin-dts'
import { nexUIRollupPlugin } from '@nex-ui/plugins'

const dirname = fileURLToPath(new URL('./', import.meta.url))

export default defineConfig([
  {
    plugins: [
      del({
        targets: './dist/*',
      }),
      typescript({
        tsconfig: path.resolve(dirname, 'tsconfig.json'),
      }),
      nexUIRollupPlugin(),
    ],
    input: './src/index.ts',
    external: [
      'react',
      'react-dom',
      'react/jsx-runtime',
      'react-dom/client',
      '@nex-ui/utils',
      '@nex-ui/css-system',
      'classnames',
      'framer-motion',
      '@iconify/react',
      '@vanilla-extract/dynamic',
      '@vanilla-extract/css',
      '@vanilla-extract/css/functionSerializer',
    ],
    output: [
      {
        preserveModules: true,
        format: 'es',
        dir: './dist/es',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
      {
        preserveModules: true,
        format: 'cjs',
        dir: './dist/lib',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
    ],
  },
  {
    plugins: [dts()],
    input: './src/index.ts',
    output: {
      dir: './dist/types',
      format: 'es',
      preserveModules: true,
    },
  },
])
