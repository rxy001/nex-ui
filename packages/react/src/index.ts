import './types/declare'

export * from './components'
export * from './theme'

export type { Theme } from './types/theme'
export type { ComponentsTheme } from './types/componentsTheme'

export { useColorScheme } from '@nex-ui/system'
export { nex, styled } from '@nex-ui/styled'
