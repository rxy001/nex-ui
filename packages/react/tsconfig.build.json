{
  "extends": "../../tsconfig.json",
  "include": ["./src"],
  "exclude": ["./src/**/__tests__/**/*", "./src/**/__stories__/**/*"]
}
