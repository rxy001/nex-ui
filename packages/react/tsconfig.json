{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "~/*": ["../../*"],
      "@theme": ["./src/theme"]
    }
  },
  "include": ["./src", "./rollup.config.ts"]
}
