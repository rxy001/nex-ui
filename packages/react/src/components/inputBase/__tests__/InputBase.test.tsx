import { useState } from 'react'
import {
  testComponentStability,
  renderWithNexUIProvider,
  testRefForwarding,
} from '~/tests/shared'
import { fireEvent } from '@testing-library/react'
import { InputBase } from '../index'

jest.mock('@nex-ui/hooks', () => {
  const originalModule = jest.requireActual('@nex-ui/hooks')

  return {
    __esModule: true,
    ...originalModule,
    useFocusRing: () => ({
      focusVisible: true,
    }),
  }
})

describe('InputBase', () => {
  afterAll(() => {
    jest.clearAllMocks()
  })
  testComponentStability(<InputBase />)

  testRefForwarding(<InputBase />)

  it('should render with default props', () => {
    const { container } = renderWithNexUIProvider(<InputBase />)

    const root = container.firstElementChild

    expect(root).toMatchSnapshot()
  })

  it('should have aria-invalid=true attribute when invalid', () => {
    const { getByRole } = renderWithNexUIProvider(<InputBase invalid />)
    expect(getByRole('textbox')).toHaveAttribute('aria-invalid', 'true')
  })

  it('should have tabIndex=-1 when disabled', () => {
    const { getByRole } = renderWithNexUIProvider(<InputBase disabled />)
    expect(getByRole('textbox')).toHaveAttribute('tabIndex', '-1')
  })

  it('should have value, disabled, placeholder, type, checked, required, readOnly and name props for input element', () => {
    const { getByRole } = renderWithNexUIProvider(
      <InputBase
        disabled
        required
        readOnly
        checked
        value='test'
        placeholder='Enter text'
        type='text'
        name='test-input'
      />,
    )

    const textbox = getByRole('textbox')

    expect(textbox).toHaveAttribute('value', 'test')
    expect(textbox).toHaveAttribute('placeholder', 'Enter text')
    expect(textbox).toHaveAttribute('type', 'text')
    expect(textbox).toHaveAttribute('name', 'test-input')
    expect(textbox).toHaveAttribute('readonly')
    expect(textbox).toHaveAttribute('required')
    expect(textbox).toHaveAttribute('disabled')
    expect(textbox).toHaveAttribute('checked')
  })

  it('should have aria-* attributes for non-input element', () => {
    const { getByRole } = renderWithNexUIProvider(
      <InputBase as='div' type='checkbox' disabled required readOnly checked />,
    )

    const checkbox = getByRole('checkbox')

    expect(checkbox).toHaveAttribute('aria-disabled', 'true')
    expect(checkbox).toHaveAttribute('aria-required', 'true')
    expect(checkbox).toHaveAttribute('aria-readonly', 'true')
    expect(checkbox).toHaveAttribute('aria-checked', 'true')
  })

  it('should correctly handle aria-* attributes of non-input element', () => {
    const { getByRole } = renderWithNexUIProvider(
      <InputBase
        as='div'
        role='checkbox'
        aria-disabled='true'
        disabled={false}
        aria-checked='true'
        checked={false}
        aria-required='true'
        required={false}
        aria-readonly='true'
        readOnly={false}
      />,
    )

    const checkbox = getByRole('checkbox')

    expect(checkbox).toHaveAttribute('aria-disabled', 'true')
    expect(checkbox).toHaveAttribute('aria-checked', 'true')
    expect(checkbox).toHaveAttribute('aria-required', 'true')
    expect(checkbox).toHaveAttribute('aria-readonly', 'true')
  })

  it('should have correct role for elements', () => {
    const { container, rerender } = renderWithNexUIProvider(
      <InputBase as='div' type='checkbox' />,
    )

    expect(container.firstElementChild).toHaveAttribute('role', 'checkbox')

    rerender(<InputBase as='div' type='radio' />)
    expect(container.firstElementChild).toHaveAttribute('role', 'radio')

    rerender(<InputBase as='div' type='text' />)
    expect(container.firstElementChild).toHaveAttribute('role', 'textbox')

    rerender(<InputBase as='div' type='date' />)
    expect(container.firstElementChild).not.toHaveAttribute('role')
  })

  it('should have data-visible attribute when focused', () => {
    const { getByRole } = renderWithNexUIProvider(<InputBase />)
    expect(getByRole('textbox')).toHaveAttribute('data-focus-visible', 'true')
  })

  it('should call onCheckedChange on click for non-input checkable elements', () => {
    const onCheckedChange = jest.fn()

    const { getByRole, rerender } = renderWithNexUIProvider(
      <InputBase as='div' role='checkbox' onCheckedChange={onCheckedChange} />,
    )

    fireEvent.click(getByRole('checkbox'))

    expect(onCheckedChange).toHaveBeenCalledTimes(1)

    rerender(
      <InputBase as='div' role='switch' onCheckedChange={onCheckedChange} />,
    )
    fireEvent.click(getByRole('switch'))
    expect(onCheckedChange).toHaveBeenCalledTimes(2)

    rerender(
      <InputBase as='div' type='radio' onCheckedChange={onCheckedChange} />,
    )
    fireEvent.click(getByRole('radio'))
    expect(onCheckedChange).toHaveBeenCalledTimes(3)
  })

  it('should call onCheckedChange on keyup for non-input checkable elements', () => {
    const onCheckedChange = jest.fn()
    const { getByRole, rerender } = renderWithNexUIProvider(
      <InputBase as='div' role='checkbox' onCheckedChange={onCheckedChange} />,
    )

    fireEvent.keyUp(getByRole('checkbox'), { key: ' ' })

    expect(onCheckedChange).toHaveBeenCalledTimes(1)

    rerender(
      <InputBase as='div' role='switch' onCheckedChange={onCheckedChange} />,
    )
    fireEvent.keyUp(getByRole('switch'), { key: ' ' })
    expect(onCheckedChange).toHaveBeenCalledTimes(2)

    rerender(
      <InputBase as='div' type='radio' onCheckedChange={onCheckedChange} />,
    )
    fireEvent.keyUp(getByRole('radio'), { key: ' ' })
    expect(onCheckedChange).toHaveBeenCalledTimes(3)
  })

  it('should call onCheckedChange on change for input checkable elements', () => {
    const onCheckedChange = jest.fn()
    const { getByRole, rerender } = renderWithNexUIProvider(
      <InputBase as='input' type='radio' onCheckedChange={onCheckedChange} />,
    )
    fireEvent.click(getByRole('radio'))
    expect(onCheckedChange).toHaveBeenCalledTimes(1)

    rerender(
      <InputBase
        as='input'
        type='checkbox'
        onCheckedChange={onCheckedChange}
      />,
    )
    fireEvent.click(getByRole('checkbox'))
    expect(onCheckedChange).toHaveBeenCalledTimes(2)
  })

  it('should not call onCheckedChange when disabled', () => {
    const onCheckedChange = jest.fn()
    const { getByRole, rerender } = renderWithNexUIProvider(
      <InputBase
        as='div'
        role='checkbox'
        checked={false}
        disabled
        onCheckedChange={onCheckedChange}
      />,
    )

    let checkbox = getByRole('checkbox')

    fireEvent.keyUp(checkbox, { key: ' ' })

    expect(onCheckedChange).not.toHaveBeenCalled()

    fireEvent.click(checkbox)

    expect(onCheckedChange).not.toHaveBeenCalled()

    rerender(
      <InputBase
        type='checkbox'
        disabled
        onCheckedChange={onCheckedChange}
        checked={false}
      />,
    )

    checkbox = getByRole('checkbox')
    fireEvent.click(checkbox)
    expect(onCheckedChange).not.toHaveBeenCalled()
  })

  it('should not call onCheckedChange when uncheckable elements', () => {
    const onCheckedChange = jest.fn()

    const { getByRole, rerender, container } = renderWithNexUIProvider(
      <InputBase
        as='div'
        role='textbox'
        checked={false}
        onCheckedChange={onCheckedChange}
      />,
    )

    const textbox = getByRole('textbox')

    fireEvent.keyUp(textbox, { key: ' ' })
    expect(onCheckedChange).not.toHaveBeenCalled()

    fireEvent.click(textbox)
    expect(onCheckedChange).not.toHaveBeenCalled()

    rerender(
      <InputBase as='div' type='text' onCheckedChange={onCheckedChange} />,
    )
    fireEvent.click(container.firstElementChild!)
    expect(onCheckedChange).not.toHaveBeenCalled()
  })

  it('should switch checked state for non-input element (uncontrolled)', () => {
    function Checkbox() {
      const [checked, setChecked] = useState(false)

      return (
        <>
          <InputBase
            as='div'
            defaultChecked={false}
            role='checkbox'
            onCheckedChange={setChecked}
          />
          <span data-testid='checkbox-value'>{`${checked}`}</span>
        </>
      )
    }

    const { getByRole, getByTestId } = renderWithNexUIProvider(<Checkbox />)

    const checkbox = getByRole('checkbox')
    const value = getByTestId('checkbox-value')

    expect(value.textContent).toBe('false')
    expect(checkbox).toHaveAttribute('aria-checked', 'false')

    fireEvent.click(checkbox)
    expect(value.textContent).toBe('true')
    expect(checkbox).toHaveAttribute('aria-checked', 'true')

    fireEvent.click(checkbox)
    expect(value.textContent).toBe('false')
    expect(checkbox).toHaveAttribute('aria-checked', 'false')

    fireEvent.keyUp(checkbox, { key: ' ' })
    expect(value.textContent).toBe('true')
    expect(checkbox).toHaveAttribute('aria-checked', 'true')

    fireEvent.keyUp(checkbox, { key: ' ' })
    expect(value.textContent).toBe('false')
    expect(checkbox).toHaveAttribute('aria-checked', 'false')
  })

  it('should switch checked state for non-input element (controlled)', () => {
    function Checkbox() {
      const [checked, setChecked] = useState(false)

      return (
        <>
          <InputBase
            as='div'
            checked={checked}
            role='checkbox'
            onCheckedChange={setChecked}
          />
          <span data-testid='checkbox-value'>{`${checked}`}</span>
        </>
      )
    }

    const { getByRole, getByTestId } = renderWithNexUIProvider(<Checkbox />)

    const checkbox = getByRole('checkbox')
    const value = getByTestId('checkbox-value')

    expect(value.textContent).toBe('false')
    expect(checkbox).toHaveAttribute('aria-checked', 'false')

    fireEvent.click(checkbox)
    expect(value.textContent).toBe('true')
    expect(checkbox).toHaveAttribute('aria-checked', 'true')

    fireEvent.click(checkbox)
    expect(value.textContent).toBe('false')
    expect(checkbox).toHaveAttribute('aria-checked', 'false')

    fireEvent.keyUp(checkbox, { key: ' ' })
    expect(value.textContent).toBe('true')
    expect(checkbox).toHaveAttribute('aria-checked', 'true')

    fireEvent.keyUp(checkbox, { key: ' ' })
    expect(value.textContent).toBe('false')
    expect(checkbox).toHaveAttribute('aria-checked', 'false')
  })

  it('should handle as for function components', () => {
    const { container } = renderWithNexUIProvider(
      <InputBase as={() => <div />} />,
    )

    expect(container.firstChild).toBeInstanceOf(HTMLDivElement)
  })

  it('should correctly handle non-input radio events', () => {
    function Radio() {
      const [checked, setChecked] = useState(false)

      return (
        <>
          <InputBase
            as='div'
            role='radio'
            checked={checked}
            onCheckedChange={setChecked}
          />
          <span data-testid='radio-value'>{`${checked}`}</span>
        </>
      )
    }

    const { getByRole, getByTestId } = renderWithNexUIProvider(<Radio />)

    const radio = getByRole('radio')

    const value = getByTestId('radio-value')

    fireEvent.click(radio)

    expect(value.textContent).toBe('true')

    fireEvent.click(radio)

    expect(value.textContent).toBe('true')

    fireEvent.keyUp(radio, { key: ' ' })

    expect(value.textContent).toBe('true')
  })
})
