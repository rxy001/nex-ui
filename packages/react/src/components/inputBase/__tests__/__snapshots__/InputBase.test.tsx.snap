// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBase should render with default props 1`] = `
.emotion-0 {
  margin: 0;
  padding: 0;
  border: none;
  background: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-radius: 0;
  box-shadow: none;
  box-sizing: border-box;
}

.emotion-0[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}

.emotion-0[type="search"]::-webkit-search-cancel-button {
  -webkit-appearance: none;
}

.emotion-0:is(:-webkit-autofill, :autofill) {
  background-color: transparent!important;
  -webkit-transition: background-color 50000s ease-in-out 0s;
  transition: background-color 50000s ease-in-out 0s;
}

<input
  class="emotion-0"
  data-focus-visible="true"
  tabindex="0"
  type="text"
/>
`;
